src/domain/accounts/domain/src/repos/account_repository.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use internal_error::InternalError;
+use opendatafabric::{AccountID, AccountName};
+use thiserror::Error;
+
+use crate::Account;
+
+///////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait AccountRepository: Send + Sync {
+    async fn create_account(&self, account: &Account) -> Result<(), CreateAccountError>;
+
+    async fn get_account_by_id(
+        &self,
+        account_id: &AccountID,
+    ) -> Result<Account, GetAccountByIdError>;
+
+    async fn find_account_id_by_email(
+        &self,
+        email: &str,
+    ) -> Result<Option<AccountID>, FindAccountIdByEmailError>;
+
+    async fn find_account_id_by_github_id(
+        &self,
+        github_id: i64,
+    ) -> Result<Option<AccountID>, FindAccountIdByGithubIdError>;
```

Comment:
The more I think about this - the more I want auth providers to be fully dynamic. If we support all providers that [Jupyter supports](https://z2jh.jupyter.org/en/stable/administrator/authentication.html#oauth2-based-authentication) this alone will put us at 8+ tables. If we add wallet auth for different blockchains - that will be even more.

Reply:
Mentioned above. I don't see a problem in having many strictly-typed tables, especially if we decouple a bit more. We could have a generic repository (find by ID, by name, email), and repositories for provider-specific data associated with accounts.